File: app/src/main/java/valiev/timur/tapfight/GameOverActivity.java
package valiev.timur.tapfight;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class GameOverActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_game_over);

        Intent intent = getIntent();

        int p1Count = intent.getIntExtra("P1_SCORE", 0);
        int p2Count = intent.getIntExtra("P2_SCORE", 0);
        System.out.println(p1Count + "/" + p2Count);

        TextView p1Score = findViewById(R.id.p1FinalScore);
        TextView p2Score = findViewById(R.id.p2FinalScore);
        TextView winner = findViewById(R.id.winnerText);

        p1Score.setText(Integer.toString(p1Count));
        p2Score.setText(Integer.toString(p2Count));

        if (p1Count > p2Count)
            winner.setText("Player 1 wins!");
        else if (p2Count > p1Count)
            winner.setText("Player 2 wins!");
    }

    public void onBackPressed() {
        startActivity(new Intent(this, StartGameActivity.class));
    }
}

File: app/src/main/java/minteemer/tapfight/presentation/gameOver/GameOverActivity.kt
package minteemer.tapfight.presentation.gameOver

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import minteemer.tapfight.R
import minteemer.tapfight.databinding.ActivityGameOverBinding
import minteemer.tapfight.util.extensions.fastLazy
import minteemer.tapfight.util.extensions.viewbinding.viewBinding

class GameOverActivity : AppCompatActivity() {

    private val binding: ActivityGameOverBinding by viewBinding(ActivityGameOverBinding::inflate)

    private val p1Score by fastLazy { intent.getIntExtra(EXTRA_P1_SCORE, 0) }
    private val p2Score by fastLazy { intent.getIntExtra(EXTRA_P2_SCORE, 0) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding.p1Score.text = p1Score.toString()
        binding.p2Score.text = p2Score.toString()

        binding.winnerText.setText(
            when {
                p1Score > p2Score -> R.string.game_over_p1_wins
                p1Score < p2Score -> R.string.game_over_p2_wins
                else -> R.string.game_over_text_draw
            }
        )
    }

    companion object {
        private const val EXTRA_P1_SCORE = "p1_score"
        private const val EXTRA_P2_SCORE = "p2_score"

        fun createIntent(context: Context, player1Score: Int, player2Score: Int) =
            Intent(context, GameOverActivity::class.java)
                .putExtra(EXTRA_P1_SCORE, player1Score)
                .putExtra(EXTRA_P2_SCORE, player2Score)
    }
}
